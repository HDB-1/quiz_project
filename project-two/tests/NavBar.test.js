import React from 'react';
import { shallow , mount } from 'enzyme';
import { MemoryRouter , Link } from 'react-router-dom';
import NavBar from '../src/components/NavBar/NavBar'

describe('Shallow NavBar', () => {
  let wrapper;
  
  beforeEach(() => wrapper = shallow(<NavBar />)); // this automatically tests for rendering without crashing.
  it("Should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  })

  it("Should render a <Link /> element", () => {
    expect(wrapper.find(Link).length).toEqual(1);
  })
  it("Should create a link to the '/' route.", () => {
      expect(wrapper.find(Link).props()).toHaveProperty("to", "/");
  })
})
