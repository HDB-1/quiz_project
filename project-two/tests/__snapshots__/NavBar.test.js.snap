// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow NavBar Should render correctly 1`] = `
<div>
  <Link
    to="/"
  >
    HOME
  </Link>
</div>
`;
